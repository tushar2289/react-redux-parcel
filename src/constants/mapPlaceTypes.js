export const RESTAURANT = 'restaurant';
export const BANK = 'bank';
export const HOSPITAL = 'hospital';
export const ATM = 'atm';
export const AIRPORT = 'airport';
export const TRAIN_STATION = 'train_station';
export const SUPERMARKET = 'supermarket';
export const GAS_STATION = 'gas_station';
export const SHOPPING_MALL = 'shopping_mall';
