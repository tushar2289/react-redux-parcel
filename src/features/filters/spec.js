import placeTypesReducer, {
  enablePlaceType,
  resetPlaceType,
} from './placeTypeSlice';
import { RESTAURANT } from '../../constants/mapPlaceTypes';

describe('Place type reducer', () => {
  let initialState;
  beforeEach(() => {
    initialState = placeTypesReducer(undefined, {});
  });

  it('should return initial state', () => {
    expect(initialState).toMatchSnapshot();
  });

  it('should set and then reset all the place types', () => {
    const newState = placeTypesReducer(initialState, {
      type: enablePlaceType.type,
      payload: {
        type: RESTAURANT,
        value: true,
      },
    });
    expect(
      placeTypesReducer(newState, {
        type: resetPlaceType.type,
      })
    ).toMatchSnapshot();
  });
});
