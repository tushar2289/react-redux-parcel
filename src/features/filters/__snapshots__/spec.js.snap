// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Place type reducer should return initial state 1`] = `
Object {
  "airport": Object {
    "isOn": false,
    "label": "Airport",
    "type": "airport",
  },
  "atm": Object {
    "isOn": false,
    "label": "ATM",
    "type": "atm",
  },
  "bank": Object {
    "isOn": false,
    "label": "Bank",
    "type": "bank",
  },
  "gas_station": Object {
    "isOn": false,
    "label": "Fuel pump",
    "type": "gas_station",
  },
  "hospital": Object {
    "isOn": false,
    "label": "Hospital",
    "type": "hospital",
  },
  "restaurant": Object {
    "isOn": true,
    "label": "Restaurant",
    "type": "restaurant",
  },
  "shopping_mall": Object {
    "isOn": false,
    "label": "Mall",
    "type": "shopping_mall",
  },
  "supermarket": Object {
    "isOn": false,
    "label": "Supermarket",
    "type": "supermarket",
  },
  "train_station": Object {
    "isOn": false,
    "label": "Train station",
    "type": "train_station",
  },
}
`;

exports[`Place type reducer should set and then reset all the place types 1`] = `
Object {
  "airport": Object {
    "isOn": false,
    "label": "Airport",
    "type": "airport",
  },
  "atm": Object {
    "isOn": false,
    "label": "ATM",
    "type": "atm",
  },
  "bank": Object {
    "isOn": false,
    "label": "Bank",
    "type": "bank",
  },
  "gas_station": Object {
    "isOn": false,
    "label": "Fuel pump",
    "type": "gas_station",
  },
  "hospital": Object {
    "isOn": false,
    "label": "Hospital",
    "type": "hospital",
  },
  "restaurant": Object {
    "isOn": true,
    "label": "Restaurant",
    "type": "restaurant",
  },
  "shopping_mall": Object {
    "isOn": false,
    "label": "Mall",
    "type": "shopping_mall",
  },
  "supermarket": Object {
    "isOn": false,
    "label": "Supermarket",
    "type": "supermarket",
  },
  "train_station": Object {
    "isOn": false,
    "label": "Train station",
    "type": "train_station",
  },
}
`;
